packages/amplify-graphql-auth-transformer/src/utils/validations.ts

```diff
@@ -94,13 +100,45 @@ found '${rule.provider}' assigned.`,
   }
 };
 
-export const validateRules = (rules: AuthRule[], configuredAuthProviders: ConfiguredAuthProviders, typeName: string) => {
+export const validateRules = (
+  rules: AuthRule[],
+  configuredAuthProviders: ConfiguredAuthProviders,
+  typeName: string,
+  context: TransformerSchemaVisitStepContextProvider,
```

Comment:
I'm a bit uncomfortable plumbing the transformer DataSourceProvider/transformer context through to this level, since we've avoided coupling up until now. But the validation needs to be able to tell if these are SQL strategies, and it's cleaner to centralize validations in a single file rather than spreading them through the transformer code, so this seemed like a compromise I could live with.